packages/schematics/angular/utility/dependency.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { Rule, SchematicContext } from '@angular-devkit/schematics';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import * as path from 'path';
+
+const installTasks = new WeakMap<SchematicContext, string[]>();
+
+interface MinimalPackageManifest {
+  dependencies?: Record<string, string>;
+  devDependencies?: Record<string, string>;
+  peerDependencies?: Record<string, string>;
+}
+
+/**
+ * Adds a package as a dependency to a `package.json`. By default the `package.json` located
+ * at the schematic's root will be used. The `manifestPath` option can be used to explicitly specify
+ * a `package.json` in different location. The type of the dependency can also be specified instead
+ * of the default of the `dependencies` section by using the `type` option with either `dev` or `peer`
+ * values.
+ *
+ * When using this rule, {@link NodePackageInstallTask} does not need to be included directly by
+ * a schematic. A package manager install task will be automatically scheduled as needed.
+ *
+ * @param name The name of the package to add.
+ * @param version The version specifier for the package to add.
+ * @param options An optional object that can contain the `type` of the dependency
+ * and/or a path (`manifestPath`) of a manifest file (`package.json`) to modify.
+ * @returns A Schematics {@link Rule}
+ */
+export function addDependency(
+  name: string,
+  version: string,
+  options: { type?: 'dev' | 'peer'; manifestPath?: string } = {},
```

Comment:
Nit: I recommend *not* calling this `manifestPath` as `package.json` is not typically thought of as "manifest" and might be confused with other things like web app manifests or browser extension manifests.

Unfortunately there's no good generic name for `package.json`, so I suggest either `packageJson` or `packageJsonPath`. Not very elegant, but I think that would be more clear.

Reply:
Yarn calls them manifest files (https://yarnpkg.com/configuration/manifest) but we can go with `packageJsonPath`.

Reply:
Interesting, I don't think I've ever heard of `package.json` files referred to as manifests, though I don't normally use Yarn.

Reply:
The libraries (e.g., https://github.com/npm/pacote#readme) used by `npm` also use the term but from a quick look the `npm` documentation doesn't appear to mention it.